### PR TITLE
fix(Popover): place arrow correctly

### LINKED ISSUE #2649: popover arrow is not anchored correctly to the popover
The arrow is in the middle of the element it is anchored to, but should be anchored to the popover.
The issue seems to be that the arrow does not adhere to what `position` is set.

See:
image

Working example: https://codesandbox.io/p/sandbox/next-digdir-designsystemet-wgr526

### PR DESCRIPTION
**Pull request by Barsnes**
resolves #2649

This check the width of both elements, and places the arrow in the center of the shortest one,

### CODE REVIEW

File: packages/react/src/components/Popover/Popover.tsx [hunk 1]
```diff
@@ -176,12 +176,30 @@ const arrowPseudoElement = {
...
+     * if popover is wider, set arrow in the middle of the reference
+     * if the reference is wider, set the arrow in the middle of the popover
+     */
+    if (rects.reference.width > rects.floating.width) {
+      arrowX = rects.floating.width / 2 - ARROW_HEIGHT / 2;
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+    } else {
+      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+    }
```
Comment (eirikbacker) [inline packages/react/src/components/Popover/Popover.tsx:192]:
```suggestion
    if (rects.reference.width > rects.floating.width) {
      arrowX = rects.floating.width / 2 - ARROW_HEIGHT / 2;
    } else {
      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
    }
    arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
```

Comment (eirikbacker) [reply packages/react/src/components/Popover/Popover.tsx:192]: Since `arrowY` is always the same? :)

Comment (mimarz) [reply packages/react/src/components/Popover/Popover.tsx:192]: Suggest also removing the `else` statement for better readability as its redundant.

Comment (Barsnes) [reply packages/react/src/components/Popover/Popover.tsx:192]: Thought I disagree with the readability, I'll move the else assignment as the default.

File: packages/react/src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -176,12 +176,43 @@ const arrowPseudoElement = {
...
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+      /**
+       * if popover is wider, set arrow in the middle of the reference
+       */
+      if (rects.reference.width > rects.floating.width) {
+        arrowX = rects.floating.width / 2 - ARROW_HEIGHT / 2;
+      }
+    } else if (placement.includes('left') || placement.includes('right')) {
+      /* default `arrowX` is if popover is bigger than the reference */
+      arrowX = rects.reference.height / 2 + rects.reference.x - data.x;
```
Comment (mimarz) [inline packages/react/src/components/Popover/Popover.tsx:196]: Is this supposed to be `height` or `width`?

Comment (mimarz) [reply packages/react/src/components/Popover/Popover.tsx:196]: Is it working as intended by chance or just a coincident? This is how you get bugs 😅

Comment (Barsnes) [reply packages/react/src/components/Popover/Popover.tsx:196]: It is working as expected, I changed this around until I got expected behaviour. Will double check and add some comments

Comment (Barsnes) [reply packages/react/src/components/Popover/Popover.tsx]: updated comments in code, did some cleaning. Still looks good

File: packages/react/src/components/Popover/Popover.tsx [hunk 3]
```diff
@@ -176,12 +176,43 @@ const arrowPseudoElement = {
...
+      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
+
+      /* default `arrowY` to center of reference, if popover is taller */
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+
+      /* if reference is taller, set arrow in the middle of the floating */
+      if (rects.reference.height > rects.floating.height) {
+        arrowY = rects.floating.height / 2;
+      }
+    }
```
Comment (mimarz) [inline packages/react/src/components/Popover/Popover.tsx:205]: Sorry I am nitpicking, but I find it hard to follow the code and values defined. `arrowX` and `arrowY` are defined with the same values inside each of the ifs, but in opposite order and contradicting comments.

I also tested without the placement checks, and works as intended still.

```suggestion
    let arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
    let arrowY = rects.reference.height / 2 + rects.reference.y - data.y;

    if (rects.reference.width > rects.floating.width) {
      arrowX = rects.floating.width / 2;
    }

    if (rects.reference.height > rects.floating.height) {
      arrowY = rects.floating.height / 2;
    }
```

**Review by mimarz — APPROVED** [review]
